src/components/Avatar/Avatar.style.ts

```diff
@@ -0,0 +1,63 @@
+import { css, SerializedStyles } from '@emotion/core';
+import { Theme } from '../../theme';
+import { rem } from 'polished';
+import { flex } from '../../theme/functions';
+import { colorShades, flatColors, pickTextColorFromSwatches } from '../../theme/palette';
+import { AvatarShapes, AvatarSizes } from './Avatar';
+
+const sizeBasedOnProp = (size: AvatarSizes) => {
```

Comment:
Why don't we use an object here? 

example:
```
const sizeBasedOnProp = {
  'sm': rem(40),
  'md': rem(46),
  'xs': rem(24),
}

sizeBasedOnProp[size] || defaultSize

```

Maybe I've missed something in the business logic here.

Reply:
same thing isn't it? also my solution offers the `default` return that in TS will be avoided by in not in JS :)

Reply:
Actually, it isn't. 

It's similar but having many `if` conditions here means that you will have more performance cost (asking every time for the correct case) than just picking the correct object's reference. 

Also, it's a more expandable way by adding a new prop with a value than adding an if condition.

Although, if your solution serves its purpose, I'm totally fine 😃 .

Reply:
i will change it to a switch case as it will serve better for our case